content/100-getting-started/02-setup-prisma/200-add-to-existing-project/110-relational-databases/150-introspection.mdx

```diff
@@ -827,6 +891,142 @@ Learn more about this on the [Configuring your Prisma Client API](/concepts/comp
+  createdAt DateTime @default(now()) @db.Timestamp(6)
+  content   String?
+  published Boolean  @default(false)
+  authorId  BigInt
+  author    User     @relation(fields: [authorId], references: [id], onDelete: NoAction, onUpdate: NoAction, map: "fk_authorId_ref_User")
+}
+
+model Profile {
+  id     BigInt  @id(map: "primary") @default(autoincrement())
+  bio    String?
+  userId BigInt  @unique
+  user   User    @relation(fields: [userId], references: [id], onDelete: NoAction, onUpdate: NoAction, map: "fk_userId_ref_User")
+}
+
+model User {
+  id      BigInt   @id(map: "primary") @default(autoincrement())
+  name    String?  @db.VarChar(255)
+  email   String   @unique @db.VarChar(255)
+  posts   Post[]
+  profile Profile?
+}
+```
+
+In this example, the database schema did follow the [naming conventions](/reference/api-reference/prisma-schema-reference#naming-conventions) <span class="api"></span> for Prisma models (only the virtual relation fields that were generated from introspection did not adhere to them and needed adjustment). This optimizes the ergonomics of the generated Prisma Client API.
+
+Sometimes though, you may want to make additional changes to the names of the columns and tables that are exposed in the Prisma Client API. A common example is to translate _snake_case_ notation which is often used in database schemas into _PascalCase_ and _camelCase_ notations which feel more natural for JavaScript/TypeScript developers.
+
+Assume you obtained the following model from introspection that's based on _snake_case_ notation:
+
+```prisma no-lines
+model my_user {
+  user_id    Int     @id @default(autoincrement())
+  first_name String?
+  last_name  String  @unique
+}
+```
+
+If you generated a Prisma Client API for this model, it would pick up the _snake_case_ notation in its API:
+
+```ts no-lines
+const user = await prisma.my_user.create({
+  data: {
+    first_name: 'Alice',
+    last_name: 'Smith',
+  },
+})
+```
+
+If you don't want to use the table and column names from your database in your Prisma Client API, you can configure them with [`@map` and `@@map`](/concepts/components/prisma-schema/data-model#mapping-model-names-to-tables-or-collections):
+
+```prisma no-lines
+model MyUser {
+  userId    Int     @id @default(autoincrement()) @map("user_id")
+  firstName String? @map("first_name")
+  lastName  String  @unique @map("last_name")
+
+  @@map("my_user")
+}
+```
+
+With this approach, you can name your model and its fields whatever you like and use the `@map` (for field names) and `@@map` (for models names) to point to the underlying tables and columns. Your Prisma Client API now looks as follows:
+
+```ts no-lines
+const user = await prisma.myUser.create({
+  data: {
+    firstName: 'Alice',
+    lastName: 'Smith',
+  },
+})
+```
+
+Learn more about this on the [Configuring your Prisma Client API](/concepts/components/prisma-client/working-with-prismaclient/use-custom-model-and-field-names) page.
```

Comment:
Off topic: I really find this section in these tutorials confusing and out of place.

Reply:
I agree, feels out of place to me too. I've made a note about this for when we get round to fixing other general problems with the getting started guides